tests/Tests/Properties/Substrings.hs

```diff
@@ -157,6 +159,48 @@ t_inits           = L.inits       `eqP` (map unpackS . T.inits)
 tl_inits          = L.inits       `eqP` (map unpackS . TL.inits)
 t_tails           = L.tails       `eqP` (map unpackS . T.tails)
 tl_tails          = L.tails       `eqPSqrt` (map unpackS . TL.tails)
+
+-- To generate function arguments for spanM:
+-- - we fix the accumulator to an Int counter, without loss of generality
+-- - we reduce the probability of producing False, by first generating an Int
+--   (here in [-5, 5]) and then mapping one outcome (0) to False and the others to True.
+genSpanFun :: Gen (Fun (Int, Char) Bool)
+genSpanFun = (fmap . fmap) (/= (0 :: Int)) (resize 5 arbitrary)
```

Comment:
I'd rather define 

```haskell
newtype SkewedBool = Skewed Bool 

instance Arbitrary SkewedBool where 
  arbitrary = frequency [(1, pure $ Skewed False), (5, pure $ Skewed True)]
```

Then `Fun (Int, Char) SkewedBool` automatically has desired distribution and you do not need boilerplate around `forAllShrink` below.